src/app/public/host-args.ts

```diff
@@ -0,0 +1,14 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  SkyuxConfigHost
+} from './config';
+
+@Injectable()
```

Comment:
Does this need to be `Injectable`? I don't see it provided anywhere.

Reply:
Along those same lines, should this be an interface instead of a class?

Reply:
I made this an injectable so that Angular wouldn't complain when I used it as a constructor param for `SkyAppConfigHost`. Since `SkyAppConfigHost` provides default values, I wanted to push the options through the constructor, instead of overwriting the entire provider.

So, instead of providing it like this (no defaults being set):
```
{
  provide: SkyAppConfigHost,
  useValue: {
    host: {...}
  }
}
```

I was shooting for this (defaults applied internally):
```
{
  provide: SkyAppConfigHost,
  useFactory() {
    return new SkyAppConfigHost({
       host: {...}
    })
  }
}
```

Let me know if I'm missing something obvious.

Reply:
Having a constructor parameter carries special meaning with Angular dependency injection, which is why Angular wants it to be `Injectable`. The fact that you're using the `useFactory` method for providing it means that you have other options for constructing your service other than going through the constructor, such as setting a property, calling a method, etc. I would take advantage of that and use one of those alternate methods (maybe through use of an `init()` method that accepts the args) so Angular doesn't try to apply its more stringent rules for constructor parameters.